src/wifi/model/periodic-dmg-wifi-scheduler.cc

```diff
@@ -0,0 +1,539 @@
+  if (info.GetAllocationFormat () == ISOCHRONOUS)
+    {
+      newDuration = GetAllocationDuration (dmgTspec.GetMinimumAllocation (), dmgTspec.GetMaximumAllocation ());
+    }
+  else if (info.GetAllocationFormat () == ASYNCHRONOUS)
+    {
+      // for asynchronous allocations, the Maximum Allocation field is reserved (IEEE 802.11ad 8.4.2.136) 
+      newDuration = dmgTspec.GetMinimumAllocation ();
+    }
+  else
+    {
+      NS_FATAL_ERROR ("Allocation Format not supported");
+    }
+
+  AllocationFieldListI allocation;
+  // Retrieve the allocation for which a modification has been requested
+  for (allocation = m_addtsAllocationList.begin (); allocation != m_addtsAllocationList.end ();)
+    {
+      if ((allocation->GetAllocationID () == info.GetAllocationID ())
+          && (allocation->GetSourceAid () == sourceAid) && (allocation->GetDestinationAid () == info.GetDestinationAid ()))
+        {
+          break;
+        }
+      else
+        {
+          ++allocation;
+        }
+    }
+
+  if (allocation == m_addtsAllocationList.end ())
+    {
+      NS_FATAL_ERROR ("Required allocation does not exist.");
+    }
+
+  uint32_t currentDuration = allocation->GetAllocationBlockDuration ();
+  NS_LOG_DEBUG ("current duration=" << currentDuration << ", new duration=" << newDuration);
+  uint32_t timeDifference;
+  if (newDuration > currentDuration)
+    {
+      NS_LOG_DEBUG ("This feature is not supported.");
+      // The request cannot be accepted; maintaining old allocation duration
+      // No need to update allocation start time and remaining DTI time 
+      status.SetFailure ();
+    }
+  else
+    {
+      NS_LOG_DEBUG ("Reduction of the duration is always allowed. Proceed to update the available slots...");
+      timeDifference = currentDuration - newDuration;
+      allocation->SetAllocationBlockDuration (newDuration);
+      status.SetSuccess ();
+
+      uint32_t startAlloc = allocation->GetAllocationStart ();
+      uint32_t endAlloc = allocation->GetAllocationStart () + allocation->GetAllocationBlockDuration () + m_guardTime;
+
+      uint16_t allocPeriod = allocation->GetAllocationBlockPeriod ();
+      if (allocPeriod != 0)
+        {
+          // If the allocation to modify is periodic, we retrieve the number of blocks
+          // and update the available slots in the DTI accordingly.
+          // TODO: update also the number of blocks if the new duration allows to 
+          // add further blocks
+          uint8_t blocks = allocation->GetNumberOfBlocks ();
+          for (uint8_t i = 0; i < blocks; i++)
+            {
+              NS_LOG_DEBUG ("Modify SP Block [" << +i << "] at " << startAlloc << " till " << endAlloc);
+              UpdateAvailableSlots (startAlloc, endAlloc, timeDifference);
+              startAlloc += allocPeriod;
+              endAlloc += allocPeriod;
+            }
+        }
+      else
+        {
+          UpdateAvailableSlots (startAlloc, endAlloc, timeDifference);
```

Comment:
No, I think this is not fundamental in the current PR. The case is separated because I have to consider periodic and non-periodic allocations distinctly. 
I am not sure if I got your point

Reply:
similarly to what i wrote in another comment, non periodic allocations can be seen as an edge case of periodic ones, with a single block, so (i think) `allocation->GetNumberOfBlocks ()` at line 487 should return 1 and thus the for loop would work for that case as well.
With the TODO at line 485 you are saying that in future works you might want to add more blocks if possible, which would then be a significant change w.r.t. the non periodic allocations (even though it could still be easily managed).
So, did you separate the non periodic case because of that, just logically, or for some other reason? You know I'm not a fan of writing the same code twice, even if it's just a single line :)

Reply:
Now I understand what you meant. I decided to update as you suggested, leaving of course the TODO there to avoid forgetting about that as well.
Since the same could be said w.r.t. `AdjustExistingAllocations` I updated also that method accordingly. As you said, for non-periodic allocations the number of blocks is set to 1 and the allocation block period to 0 (it is easily verifiable in `DmgWifiScheduler`) so our solution covers all the possible conditions. 
Thanks for raising this point!